File: POOBanca/C++/Jornal.cpp
#include "Jornal.h"

void Jornal::apresentacao() {
    Produto::apresentacao();

    cout << "O jornal é diário? " << (this->diario ? "Sim" : "Não") << "." << endl;
}

Jornal::Jornal(string nome, string editora, int totPaginas, bool diario) : Produto(nome, editora, totPaginas) {
    this->diario = diario;
}

Jornal::~Jornal() {
    cout << "Objeto Jornal Destruído com Sucesso!" << endl;
}

bool Jornal::getDiario() {
    return this->diario;
}

void Jornal::setDiario(bool diario) {
    this->diario = diario;
}

File: POOBanca/Python/Jornal.py
from Produto import Produto

class Jornal(Produto):
    __diario = False

    def apresentacao(self):
        super().apresentacao()

        print("O jornal é diário? {}.".format("Sim" if self.__diario else "Não"))

    def __init__(self, nome, editora, totPaginas, diario):
        super().__init__(nome, editora, totPaginas)

        self.__diario = diario

    def __del__(self):
        print("Objeto Jornal Destruído com Sucesso!")

    def getDiario(self):
        return self.__diario

    def setDiario(self, diario):
        self.__diario = diario